File: solutions/0645-set-mismatch/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function(nums) {
    let sum = 0;
    let dnum = 0;
    let seen = new Set();

    for (let i = 0; i < nums.length; i++) {
        sum += nums[i];
        if (seen.has(nums[i])) {
            dnum = nums[i];
        }
        seen.add(nums[i]);
    }

    let n = nums.length;
    let actualSum = (n * (n + 1)) / 2;
    let misnum = actualSum - sum + dnum;

    return [dnum, misnum];
};

File: solutions/0645-set-mismatch/solution.java
class Solution {
    public int[] findErrorNums(int[] nums) {
      int n = nums.length;
        int[] count = new int[nums.length+1];
        int[] ans = new int[2];

        
        for (int num : nums) {
            count[num]++;
        }

        for (int i = 1; i <= n; i++) {
            if (count[i] == 2) {
                ans[0] = i;   
            }
            if (count[i] == 0) {
                ans[1] = i;  
            }
        }

        return ans;
    }
}
